identity_iota_core/src/state_metadata/document.rs

```diff
@@ -93,26 +95,47 @@ impl StateMetadataDocument {
       identity_did::Error::InvalidDocument("expected data to have at least length 5", None),
     )?)?;
 
-    let inner: &[u8] = data.get(5..).ok_or(identity_did::Error::InvalidDocument(
-      "expected data to have at least length 6",
-      None,
-    ))?;
+    let data_len_packed: [u8; 2] = data
+      .get(5..=6)
+      .ok_or(identity_did::Error::InvalidDocument(
+        "expected data to have at least length 8",
+        None,
+      ))?
+      .try_into()
+      .map_err(|_| identity_did::Error::InvalidDocument("failed to get data length", None))?;
+    let data_len: u16 = u16::from_le_bytes(data_len_packed);
+
+    let data: &[u8] = data
+      .get(7..7 + data_len as usize)
```

Comment:
Nit: Decrease ambiguity.

```suggestion
      .get(7..(7 + data_len as usize))
```